pkg/controller/gpudevice/vgpu_controller.go

```diff
@@ -0,0 +1,357 @@
+		// is reconfigured post reboot
+		if vGPUExists := containsVGPU(v, vGPUDevices); !parentDeviceEnabled && vGPUExists == nil {
+			if err := h.vGPUClient.Delete(v.Name, &metav1.DeleteOptions{}); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func containsVGPU(vgpu *v1beta1.VGPUDevice, vgpuList []*v1beta1.VGPUDevice) *v1beta1.VGPUDevice {
+	for _, v := range vgpuList {
+		if vgpu.Name == v.Name {
+			return v
+		}
+	}
+	return nil
+}
+
+// enableVGPU performs the op to configure VGPU
+func (h *Handler) enableVGPU(vgpu *v1beta1.VGPUDevice) (*v1beta1.VGPUDevice, error) {
+	nvidiaType, ok := vgpu.Status.AvailableTypes[vgpu.Spec.VGPUTypeName]
+	if !ok {
+		return vgpu, fmt.Errorf("VGPUType specified %s is not available for vGPU %s", vgpu.Spec.VGPUTypeName, vgpu.Spec.Address)
+	}
+
+	vgpuUUID := uuid.NewString()
+
+	createFilePath := filepath.Join(v1beta1.MdevBusClassRoot, vgpu.Spec.Address, v1beta1.MdevSupportTypesDir, nvidiaType, "create")
+	if _, err := os.Stat(createFilePath); err != nil {
+		return vgpu, fmt.Errorf("error looking up create file for vgpu %s: %v", vgpu.Name, err)
+	}
+
+	if err := os.WriteFile(createFilePath, []byte(vgpuUUID), fs.FileMode(os.O_WRONLY)); err != nil {
+		return vgpu, fmt.Errorf("error writing to create file for vgpu %s: %v", vgpu.Name, err)
+	}
+
+	vgpu.Status.VGPUStatus = v1beta1.VGPUEnabled
+	vgpu.Status.UUID = vgpuUUID
+	vgpuObj, err := h.vGPUClient.UpdateStatus(vgpu)
+	if err != nil {
+		return vgpuObj, err
+	}
+
+	return h.reconcileDisabledVGPUStatus(vgpuObj)
+}
+
+// disableVGPU performs the op to disable VGPU
+func (h *Handler) disableVGPU(vgpu *v1beta1.VGPUDevice) (*v1beta1.VGPUDevice, error) {
+	removeFile := filepath.Join(v1beta1.MdevBusClassRoot, vgpu.Spec.Address, vgpu.Status.UUID, "remove")
+	var notFound bool
+	// possible that CRD update fails but file has been removed
+	// this can lead to issue during reconcile.
+	// in such a case we just ensure plugin is updated and CRD status reflects disabled state
+	if _, err := os.Stat(removeFile); err != nil {
+		if os.IsNotExist(err) {
+			notFound = true
+		} else {
+			return vgpu, fmt.Errorf("error looking up remove file for vgpu %s: %v", vgpu.Name, err)
+		}
+	}
+
+	if !notFound {
+		if err := os.WriteFile(removeFile, []byte("1"), fs.FileMode(os.O_WRONLY)); err != nil {
+			return vgpu, fmt.Errorf("error writing to remove file for vgpu %s: %v", vgpu.Name, err)
+		}
+	}
+
+	// disableDevicePlugin is run here as we need UUID to remove the device
+	if err := h.disableDevicePlugin(vgpu); err != nil {
+		return vgpu, fmt.Errorf("error cleaning up device plugin for device %s: %v", vgpu.Name, err)
+	}
+
+	vgpu.Status.VGPUStatus = v1beta1.VGPUDisabled
+	vgpu.Status.UUID = ""
+	vgpu.Status.ConfiguredVGPUTypeName = ""
+	vgpuObj, err := h.vGPUClient.UpdateStatus(vgpu)
+	if err != nil {
+		return vgpuObj, err
+	}
+
+	return h.reconcileDisabledVGPUStatus(vgpuObj)
+}
+
+func (h *Handler) patchKubevirtCR() error {
+	return nil
+}
```

Comment:
This function seems not used.

Reply:
good spot. I had moved the logic to another method.